developers/arbitrum-deploy.md

```diff
@@ -1,121 +1,498 @@
-sudo tar -C /usr/local -xzf "go$ver.linux-amd64.tar.gz"
-rm "go$ver.linux-amd64.tar.gz"
-echo "export PATH=$PATH:/usr/local/go/bin:$HOME/go/bin" >> $HOME/.bash_profile
-source $HOME/.bash_profile
-go version
+celestia light auth admin --p2p.network mocha
 ```
 
-### Node
+<!-- markdownlint-disable MD033 -->
 
-```bash
-curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.1/install.sh | bash
-export NVM_DIR="$([ -z "${XDG_CONFIG_HOME-}" ] && printf %s "${HOME}/.nvm" || printf %s "${XDG_CONFIG_HOME}/nvm")"
-[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh" # This loads nvm
-nvm install 16.20.0
-nvm use 16.20.0
-node --version
-npm install --global yarn
-yarn --version
+Since the contracts deployed through the factories above are already configured
+to communicate with Blobstream, you now only have to configure your node
+accordingly. First understand the different variables that will be set in the config:
+
+- **`enable`:** self explanatory, set it to true if you are using Celestia DA 😁
+- **`rpc`:** RPC endpoint for **celestia-node**
+- **`tendermint-rpc`:** a celestia-core endpoint from a full node
+(**NOTE:** only needed for a batch poster node)
+- **`eth-rpc`:** Ethereum Client RPC endpoint, only used when the node is a batch poster
+- **`namespace-id`:** namespace being used to post data to Celestia
+- **`auth-token`:** auth token for your Celestia Node
+- **`is-poster`:** is the node with Celestia DA the batch poster, set to true if so.
+- **`gas-price`:** how much to pay for gas (in uTIA)
+- **`event-channel-size`:** size of the events channel used by the batch poster
+to wait for a range of headers that contains the header for the block in which
+it posted a blob, before posting the batch to the base layer for verification
+on Blobstream X.
+- **`blobstreamx-address`:** address of the Blobstream X contract on the base chain.
+    - Note that the `SequencerInbox` contract for each chain has a constant
+    address for the `BlobstreamX` contract, thus make sure that the Blobstream X
+    address in the `SequencerInbox` being used for the templates in
+    `RollupCreator` matches the one in your config.
+
+Now enable Celestia DA in your Arbitrum chain params in
+`config/nodeConfig.json`. If you'd like to use your own namespace,
+use a custom 10 byte value or random value using
+`openssl rand -hex 10` for `namespace-id`:
+
+:::warning
+The Orbit contracts depend on
+[the existing Blobstream X deployments](#blobstream-x-contract-deployments).
+Before using these addresses, **please verify the contract addresses** on
+[the official source below](#blobstream-x-contract-deployments)
+to avoid any issues due to incorrect addresses.
+This is crucial to protect against potential misuse by copy-paste errors.
+:::
+
+```ts
+"celestia-cfg": {
+  "enable": true,
+    "rpc": "http://host.docker.internal:26658",
+    "tendermint-rpc": "http://consensus-full-mocha-4.celestia-mocha.com:26657",
+    "eth-rpc": "<YOUR_ETH_RPC>",
```

Comment:
is this a wss or https url @Ferret-san ?

Reply:
the eth_rpc must be wss. Otherwise it won't be able to subscribe to events for Blobstream

Reply:
https://github.com/celestiaorg/docs/pull/1490/commits/349abff2dde339dd440d849942ab9c22b476445d

Reply:
https://github.com/celestiaorg/docs/pull/1490/commits/db635d5cddaf82811bfeb295c1eac50d21fa2b35